src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -359,6 +359,27 @@ public async Task FailedConnect_ConnectedReturnsFalse()
 
             Assert.False(socket.Connected);
         }
+
+        [Fact]
+        public async Task FailedConnect_ConnectedReturnsFalse_TimeSpan()
+        {
+            TimeSpan timeSpan = TimeSpan.FromMilliseconds(5_000_000L * 1000);
```

Comment:
Why 1,388 hours particularly? If you want a "reasonable time for any test to take before we fail it because it hung" then 30 seconds is common. If you want "time that is self evidently intended to be infinite" then use TimeSpan.MaxValue.

Reply:
I see `TimeSpan.FromSeconds(30))` above.